test/unit/feirari/AutoRewardsDistributor.test.ts

```diff
@@ -0,0 +1,158 @@
+import { expectRevert, balance, getAddresses, getCore } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers, artifacts } from 'hardhat';
+import { BigNumber, Signer } from 'ethers';
+import { join } from 'path/posix';
+
+const toBN = ethers.BigNumber.from;
+
+const MockTribalChief = artifacts.readArtifactSync('MockTribalChief');
+const MockRewardsDistributor = artifacts.readArtifactSync('MockRewardsDistributor');
+
+const RewardsDistributorAdmin = artifacts.readArtifactSync('RewardsDistributorAdmin');
+const AutoRewardsDistributor = artifacts.readArtifactSync('AutoRewardsDistributor');
+
+describe('AutoRewardsDistributor', function () {
+  let userAddress: string;
+  let minterAddress: string;
+  let pcvControllerAddress: string;
+  let governorAddress: string;
+  const e18 = '000000000000000000';
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [
+      addresses.userAddress,
+      addresses.secondUserAddress,
+      addresses.minterAddress,
+      addresses.pcvControllerAddress,
+      addresses.governorAddress,
+      addresses.genesisGroup,
+      addresses.beneficiaryAddress1,
+      addresses.beneficiaryAddress2,
+      addresses.guardianAddress,
+      addresses.burnerAddress
+    ];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress, pcvControllerAddress, governorAddress, minterAddress } = await getAddresses());
+
+
+    this.poolIndex = 1000;
+    this.poolAllocPoints = 1000;
+    this.totalAllocPoint = 10000;
+    this.tribePerBlock = '75' + e18;
```

Comment:
Never use plus when you're trying to do string concatenation; javascript has some VERY strange edge cases for this.

Instead use backtick/string substitution notation:

`this.tribePerBlock = \` 75${e18} \``

Reply:
fixed